__tests__/app.test.js

```diff
@@ -1,41 +1,80 @@
+const seed = require("../db/seeds/seed");
+const db = require("../db/connection");
 
 beforeEach(() => {
-    return seed(data);
-})
+  return seed(data);
+});
 
 afterAll(() => {
-    return db.end();
-})
+  return db.end();
+});
 
 describe("GET /api", () => {
-    test("200: Responds with an object detailing the documentation for each endpoint", () => {
-        return request(app)
-            .get("/api")
-            .expect(200)
-            .then(({body: {endpoints}}) => {
-                expect(endpoints).toEqual(endpointsJson);
-            });
-    });
+  test("200: Responds with an object detailing the documentation for each endpoint", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body: { endpoints } }) => {
+        expect(endpoints).toEqual(endpointsJson);
+      });
+  });
 });
 
-describe('GET /api/topics', () => {
-    it('should respond with status 200 and an array of topic objects', () => {
-        return request(app).get("/api/topics").expect(200).then(({body: {topics}}) => {
-            topics.forEach((topic) => {
-                expect(topic).toHaveProperty("slug")
-                expect(topic).toHaveProperty("description");
-                expect(topic).toMatchObject({
-                    description: expect.any(String),
-                    slug: expect.any(String),
-                })
-            })
-        })
-    });
-});
\ No newline at end of file
+describe("GET /api/topics", () => {
+  it("should respond with status 200 and an array of topic objects", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body: { topics } }) => {
+        topics.forEach((topic) => {
+          expect(topic).toHaveProperty("slug");
+          expect(topic).toHaveProperty("description");
+          expect(topic).toMatchObject({
+            description: expect.any(String),
+            slug: expect.any(String),
+          });
+        });
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id", () => {
+  it("should return status 200 and an article object", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body: { article } }) => {
+        expect(article).toMatchObject({
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: expect.any(String),
+          votes: 100,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
+  });
+  it("should respond with 404 not found if the article does't exist", () => {
+    return request(app)
+      .get("/api/articles/0")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Not found");
+      });
```

Comment:
0 is an interesting choice, but I would recommend changing this to be something more similar to `/api/articles/999`. This is because 0 isnt naturally be an id that would exist, but 999 could be and in this instance because it does not we expect a 404 status code and msg exactly like you have done

Reply:
yeah, my idea behind this was as follows:
Stephen said the other day that article 999 might not exist now, but might exist at some point in the future, but 0 will never exist, because sql starts counting from 1, hence why i decided to go with 0 - it'll never be present so it will always pass the test 😆